bokehjs/src/lib/core/util/serialization.ts

```diff
@@ -194,6 +194,32 @@ export type Shapes = {[key: string]: Shape | Shape[]}
 
 export type EncodedData = {[key: string]: NDArray | Arrayable}
 
+
+function traverse_data(v: any, buffers: [any, any][]): [Arrayable, any] {
+  // v is just a regular array of scalars
+  if (v.length == 0 || !(isObject(v[0]) || isArray(v[0]))) {
+    return [v, v.length]
```

Comment:
@bryevdv, is there a reason why we weren't setting `shape` to `v.length` in this case where `v` is an array?

Reply:
The original code that only handled 1d arrays just didn't require it (checked the length directly) So it simply was not added due to lack of necessity. I am not opposed to adding it to make things consistent, but let's make an issue and do that after 1.0 since I can't say with certainty that nothing ever conditions on the presence or absence of `shape` to deduce array dimension, and adding it now might beak such paths.